components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/claims/impl/DefaultClaimHandler.java

```diff
@@ -129,22 +129,29 @@ protected Map<String, String> handleFederatedClaims(Map<String, String> remoteCl
                                                         StepConfig stepConfig, AuthenticationContext context)
             throws FrameworkException {
 
-        ClaimMapping[] idPClaimMappings = context.getExternalIdP().getClaimMappings();
+        ApplicationConfig appConfig = context.getSequenceConfig().getApplicationConfig();
+        AuthenticatedUser authenticatedUser = getAuthenticatedUser(stepConfig, context);
+        String tenantDomain = authenticatedUser.getTenantDomain();
+        UserRealm realm = getUserRealm(tenantDomain);
+        if (realm == null) {
+            log.warn("No valid tenant domain provider. No claims returned back");
+            return new HashMap<>();
+        }
+        ClaimManager claimManager = getClaimManager(authenticatedUser.getTenantDomain(), realm);
 
+        ClaimMapping[] idPClaimMappings = context.getExternalIdP().getClaimMappings();
         if (idPClaimMappings == null) {
             idPClaimMappings = new ClaimMapping[0];
         }
 
-        Map<String, String> spClaimMappings = context.getSequenceConfig().getApplicationConfig().
-                getClaimMappings();
-
-        if (spClaimMappings == null) {
-            spClaimMappings = new HashMap<>();
-        }
+        boolean useDefaultClaimMappings = canUseDefaultClaimMappings(appConfig, context);
+        Map<String, String> spClaimMappings = getSpToLocalClaimMappings(useDefaultClaimMappings, appConfig,
+                claimManager);
+        Map<String, String> spRequestedClaimMappings = getRequestedClaimMappings(useDefaultClaimMappings, appConfig,
+                spClaimMappings);
```

Comment:
add useDefaultClaimMappings variable initialization inside the pvt methods.

Reply:
fixed with https://github.com/wso2/carbon-identity-framework/pull/4244/commits/4f2be12a856e0ec20b5c0fc918707a7bdc4592c0